rfcs/0013-redesign-assets-id-system.md

```diff
@@ -0,0 +1,246 @@
+# Redesign Assets System
+
+This RFC will serve to finalize our approach to refactoring assets-registry and 
+removing assets, currency-factory, and our dependence on hard-coded assets.
+
+## Context
+
+Previously, we planned to use currency-factory to reserve asset IDs. However,
+currency-factory itself was mostly difficult to maintain legacy code with an
+asset ID range system that was overly complex and not helpful to our current 
+tech stack.
+
+As it stands, we have removed most of the dependence on currency-factory from
+Picasso in favor of hard-coded asset IDs. These hard-coded IDs have allowed us
+to conduct an initial release using a simplistic storage migration to create
+initial pools for Pablo. However, with the dependence on these hard-coded asset
+IDs - we now must conduct additional, more complex, storage migrations to add 
+more Pablo pools and more tokens.
+
+Instead of depending on these hard-coded asset IDs, we should instead develop
+a dynamic ID system that makes it easy for developers to both interact with 
+assets within our consensus system and to retrieve information relevant to a 
+given asset.
+
+## Requirements
+
+* There exists a double mapping between multi-locations and local asset IDs
+  * Zero or One multi-locations ↔ One asset ID
+
+* Given a way to uniquely identify an asset, one should be able to retrieve 
+asset metadata (ticker-symbol, decimal precision, and ratio)
+
+* Support the traits from `frame_support::traits::tokens::fungibles`
+  * Specifically `MutateHold`
+
+* Ability to mint mintable assets and the inability to mint non-mintable assets
+
+* Support the existing local asset IDs we have hard-coded
+
+* Utilize both pallet-balances and pallet-assets from Parity for maintaining 
```

Comment:
This by it self is not a requirement, just that it's possibly the best approach given another constraint. Long terms compatibility between our chain and parity tools(Eg: UI) and conventions. I would avoid adding specific solutions in requirements.